File: app/src/main/java/com/example/imagecheckboxapp/MainActivity.java
package com.example.imagecheckboxapp;

import android.support.v7.app.AppCompatActivity;
import android.os.Bundle;
import android.view.View;
import android.widget.CheckBox;
import android.widget.ImageView;
import android.widget.RadioButton;
import android.widget.RadioGroup;
import android.widget.TextView;

public class MainActivity extends AppCompatActivity {
    CheckBox chk;
    ImageView imgMina,imgMomo,imgSana,imgTzuyu;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
    }

    public void btnOK(View view) {
        TextView txvLike = findViewById(R.id.txvLike);
        RadioGroup kGirlGroup = findViewById(R.id.kGirlGroup);

        int id = kGirlGroup.getCheckedRadioButtonId();
        RadioButton selected = findViewById(id);

        txvLike.setText("我的最愛是 :" + selected.toString());
    }

}

File: app/src/main/java/com/example/imagecheckboxapp/MainActivity.kt
package com.example.imagecheckboxapp

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import kotlinx.android.synthetic.main.activity_main.*

class MainActivity : AppCompatActivity() {
    CheckBox chk;
    ImageView imgMina,imgMomo,imgsana,imgTzuyu;

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        imgMina = findViewById(R.id.imgMina);
        imgMomo = findViewById(R.id.imgMomo);
        imgSana = findViewById(R.id.imgSana);
        imgTzuyu = findViewById(R.id.imgTzuyu);

        imgMina.setVisibility(View.GONE);
        imgMomo.setVisibility(View.GONE);
        imgSana.setVisibility(View.GONE);
        imgTzuyu.setVisibility(View.GONE);
    }

    fun btnOK(view: View) {

        imgMina.setVisibility(View.GONE);
        imgMomo.setVisibility(View.GONE);
        imgSana.setVisibility(View.GONE);
        imgTzuyu.setVisibility(View.GONE);

        int[] id ={R.id.chkMina,R.id.chkMomo,R.id.chkSana,R.id.chkTzuyu};

        for(int i:id){

            chk = findViewById(i);

            if(chk.isChecked()){
                if(chk.getId() == R.id.chkMina)
                    imgMina.visibility(View.VISIBLE);
                if(chk.getId() == R.id.chkMomo)
                    imgMomo.visibility(View.VISIBLE);
                if(chk.getId() == R.id.chkSana)
                    imgSana.visibility(View.VISIBLE);
                if(chk.getId() == R.id.chkTzuyu)
                    imgTzuyu.visibility(View.VISIBLE);

            }


        }

    }
}
